src/rgw/rgw_tag.h

```diff
@@ -0,0 +1,39 @@
+#ifndef RGW_TAG_H
+#define RGW_TAG_H
+
+#include <map>
+#include <string>
+#include <include/types.h>
+
+#include "rgw_common.h"
+
+class RGWObjTags
+{
+ protected:
+  std::map <std::string, std::string> tags;
```

Comment:
Well currently we are only consuming the tags in full, but in the future we may end up needing that for checking if a specific key val exists in an object, I have moved to flat_maps locally and it seems working, will push an update